distribution/tools/keystore-cli/src/test/java/org/elasticsearch/common/settings/CreateKeyStoreCommandTests.java

```diff
@@ -53,48 +53,67 @@ public void testNotMatchingPasswords() throws Exception {
     }
 
     public void testDefaultNotPromptForPassword() throws Exception {
+        assumeFalse("Cannot open unprotected keystore on FIPS JVM", inFipsJvm());
```

Comment:
Ideally, when not firefighting, I think we should assert the behavior from the FIPS JVM in circumstances such as this one. For regular JVMs, the tests assert all kinds of behaviors, but since some are skipped in the FIPS JVM we condone unspecified behavior.
@jkakavas Is https://github.com/elastic/elasticsearch/pull/65464 supposed to handle some/all cases like this one?

Granted, this assumes the FIPS provider implementation (different ones might fail differently), and maybe the JVMs as well. I see this as an argument for more clearly specifying and documenting the FIPS environments we test (and support).

Should I open a separate GH issue regarding negative testing in the FIPS JVM?

Reply:
This is not covered in 65454. We can add tests for certain cases with assumeTrue in fips mode, but only for ones we can anticipate the behavior, i.e. short passwords. As you say its tricky. 


>  I see this as an argument for more clearly specifying and documenting the FIPS environments we test (and support).

Agreed, there is an ongoing separate discussion around this

Reply:
If there are specific cases in this PR where it feels like we're skipping a test for something that should work on FIPS, then definitely point it out because I tried hard to resolve all of those (which increased coverage on non-FIPS as well, since there were scenarios that we never tested with a password).

However, in this case, this is simply something that is not supported on FIPS. The test is that we don't prompt for a password if there isn't one, and that's not a valid scenario to test of FIPS. We simply cannot read a keystore without a password.
I _could_ write a test to check that unprotected keystores fail on FIPS, except that's not really a _requirement_ per se, it's just a consequence of the JVM being in FIPS mode.

One could make an argument that we should have a test `testThatReadingUnprotectedKeystoreInFipsModeFailsWithUsefulMessageAndDoesntCorruptKeystore` (maybe a shorter test name) but it's probably not a super high priority.